buildroot-external/scripts/rauc.sh

```diff
@@ -19,10 +32,19 @@ function write_rauc_config() {
 
 
 function install_rauc_certs() {
-    if [ "${DEPLOYMENT}" == "production" ]; then
-        cp "${BR2_EXTERNAL_HASSOS_PATH}/ota/rel-ca.pem" "${TARGET_DIR}/etc/rauc/keyring.pem"
-    else
+    local cert="/build/cert.pem"
+
+    if [ "${DEPLOYMENT}" == "development" ]; then
         cp "${BR2_EXTERNAL_HASSOS_PATH}/ota/dev-ca.pem" "${TARGET_DIR}/etc/rauc/keyring.pem"
+
+        # Add local self-signed certificate (if not trusted by chain it is a
+        # self-signed certificate)
+        if ! openssl verify -CAfile "${BR2_EXTERNAL_HASSOS_PATH}/ota/dev-ca.pem" -no-CApath "${cert}"; then
```

Comment:
We use `DEPLOYMENT` only to choose which CA/keyring should be deployed to the target. I'd prefer if I could do a "development build" of any commit without changing anything - now I need to edit the `meta` file if I want to have a self-signed build from the `main` or `rc` branch.

Reply:
Technically, we could add the self-signed certificate in any case. I think the main reason I did it the way it is now was to prevent accidentally leak the self-signed certificate into a release... 

However, there is also the thought of same source -> same binary... If we change things in the image based on local environment (which certificate), the build is not really reproducible. In a way, I think it is good to fail if you can't rebuild it 100% the same way :man_shrugging:

Reply:
>  to prevent accidentally leak the self-signed certificate into a release...

That's the point of the other comment. If we avoid that by not creating the certificate in the build/release pipeline, it should be fine.

>  If we change things in the image based on local environment (which certificate), the build is not really reproducible. In a way, I think it is good to fail if you can't rebuild it 100% the same way 🤷‍♂️

But why the `dev` branch should be an exception? Right now we're just making it more complicated if you want to bisect issues on other branches.

Reply:
If you REALLY want to be sure it doesn't happen by accident, could we maybe add a possibility to override the behavior using an env variable, or showing and interactive prompt?

Reply:
> But why the `dev` branch should be an exception? Right now we're just making it more complicated if you want to bisect issues on other branches.

IMHO, a releases build and a dev/nightly build is different in the sense that for a release set/bump the version in source (to make it reproducible), whereas for dev/nightly builds the version number usually generated... So I backed this by that logic.

But yeah, it hurts bisectability. I've changed to always add the self-signed certificate, no matter the build type :+1: